docs/CPS/CPS-XXXX/README.md

```diff
@@ -0,0 +1,102 @@
+---
+CPS:
+Title: Sub-Optimal Provider to Wallet API
+Status: Open
+Category: Wallets
+Authors:
+    - Rhys Bartels-Waller <rhys.bartelswaller@iohk.io>
+    - William Wolff <william.wolff@iohk.io>
+    - Martynas Kazlauskas <martynas.kaslauskas@iohk.io>
+    - Daniele Ricci <daniele.ricci@iohk.io>
+    - Sefa Irken <sefa.irken@iohk.io>
+
+
+Proposed Solutions: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/tree/master/CPS-0011
+Created: 2024-07-09
+License: CC-BY-4.0
+---
+
+## Abstract
+The event-based nature of a Blockchain is well-suited for wallets to derive their internal state from genesis, however, most data services only expose a request/response interface and/or take on the responsibility of decoding the native binary data into JSON, enforcing a strong client/server contract with the current protocol transaction format. Wallets must use long polling to detect changes, on an interval that balances latency with network and service demand, then make numerous calls against various endpoints that have the potential to change on each protocol upgrade. Furthermore, the lack of an API standard inhibits development and adoption of a standalone desktop service for wallets to offer a full node experience for their users, which is best achieved via a universal API running either on a remote server, in a multi-tenant mode, or locally running in single-tenant mode.
+
+## Problem
+> [!NOTE]
+>
+> Fundamental wallet functionality is defined in this document as being able to construct and submit any transaction supported by the latest Cardano protocols, requiring:
+> - UTxO set owned by addresses derived within the 1852 purposed, [BIP32-ed25519](https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf) accounts or the product of hashing a script.
+> - Active stake pools
+> - Reward account balances
+> - Certificates submitted by the wallet
+> - Current Cardano network protocol parameters
+> - Era summaries, for time conversions
+> - Current chain tip
+
+### Impedance Mismatch
+- The current APIs available to light wallets tend to be part of a general purpose solution serving many application types, built on top of [Cardano DB Sync] with a request/response interaction style. To achieve a reactive UX and stay in sync with the network state, long polling is required to reconstruct the original event-based nature of the system.
```

Comment:
Not sure I understand this correctly, please elaborate :thinking: I believe many wallets are not re-constructing, but fetching current state that was re-constructed in the backend
```suggestion
- The current APIs available to light wallets tend to be part of a general purpose solution serving many application types, built on top of [Cardano DB Sync] with a request/response interaction style. To achieve a reactive UX and stay in sync with the network state, long polling is required to fetch the current state of the system and check whether there are any updates.
```

Reply:
https://github.com/will-break-it/wallet-architecture/pull/3/commits/d188be9a67146d3724a43dca709040aa376ef27b